### DISCUSSION TITLE
Support for discord webhooks?

### DISCUSSION THREAD

**Discussion by OmgImAlexis** [thinkst/canarytokens#245] [Feature Requests]
I edited the field before taking the screenshot to obfuscate the URL.

![image](https://user-images.githubusercontent.com/6525926/131193026-4dc7ec3d-4695-4f66-9b49-9ce749b07179.png)

**Comment by NDevTK** [comment]
https://discord.com/developers/docs/resources/webhook#execute-webhook

Maybe it should allow custom JSON currently webhooks are required to handle:
```
{
    "manage_url": "http://example.com/test/url/for/webhook",
    "memo": "Congrats! The newly saved webhook works",
    "additional_data": {
        "src_ip": "1.1.1.1",
        "useragent": "Mozilla/5.0...",
        "referer": "http://example.com/referrer",
        "location": "http://example.com/location"
    },
    "channel": "HTTP",
    "time": "2021-08-28 20:39:38"
}
```

**Comment by jtagcat** [comment]
The project can support [apprise](https://github.com/caronc/apprise), a library currently supporting 64 backends.

Works using an 'Apprise URL':
 - Telegram: `tgram://123456789:abcdefg_hijklmnop/12315544/`
 - Discord: `discord://4174216298/JHMHI8qBe7bk2ZwO5U711o3dV_js?avatar_url=https://i.imgur.com/FsEpmwg.jpeg`

This is maybe even compatible with the current text-based e-mail/webhook input. If it's not an e-mail address, and the protocol isn't http or https, use apprise.

**Comment by NDevTK** [comment]
Looks nice as long as the service is safe from SSRF.

**Comment by ghost** [comment]
Any updates on this?

**Comment by NDevTK** [comment]
You can use Vercel or Cloudflare workers to forward the webhook for free.
Does not seem like theirs much interest in adding the feature.

**Comment by jayjb** [comment]
Hi folks,

Sorry for the delay. I've added this as an enhancement but it seems unlikely to be a priority. I think a nice workaround here would be to create an AWS lambda that receives the information from Canarytokens server via webhook and forwards the information to discord. The lambda could run the discord library and format it correctly
